agent/consul/state/catalog.go

```diff
@@ -2859,16 +2864,20 @@ func (s *Store) ServiceTopology(
 
 	upstreamDecisions := make(map[string]structs.IntentionDecisionSummary)
 
-	// The given service is the source relative to upstreams
-	sourceURI := connect.SpiffeIDService{
+	matchEntry := structs.IntentionMatchEntry{
 		Namespace: entMeta.NamespaceOrDefault(),
-		Service:   service,
+		Name:      service,
+	}
+	// The given service is a source relative to its upstreams
+	_, intentions, err := s.IntentionMatchOne(ws, matchEntry, structs.IntentionMatchSource)
```

Comment:
Should this be calling a function that accepts the active transaction instead of creating a new transaction?

Same question about the call to `IntentionDecision` just below this.

Reply:
Yep that's right. Fixed that.

IntentionDecision doesn't need one because it doesn't query the state store anymore.